src/com/cft/contactmerge/FirstNameMatcher.java

```diff
@@ -0,0 +1,42 @@
+/* Created by James Page on May 8th, 2018
+ * Contains the order in which the first names will be considered a match or not...
+ * Exact Match (and matched with punctuations), matched with hyphenated parts, matched with apostrophe, and matched with initial.
+ */
+
+package com.cft.contactmerge;
+
+import java.util.ArrayList;
+
+public abstract class FirstNameMatcher extends NameMatchLogic {
+    public static AnswerType getFirstNameMatchResult(ArrayList<String> contactOneFirstName, ArrayList<String> contactTwoFirstName){
+        //Note: if both names contained punctuations and were split into an array of parts of a name...
+        //      and both names containing their own parts are exactly the same...
+        //      the exactMatch will be confirmed
+        if (isNamesExactMatch(contactOneFirstName, contactTwoFirstName)){
+            return AnswerType.yes;
+        }
+
+        /* Check if both names are split based on punctuation, and these names parts are in reversing/mixed order
+        if ((contactOneFirstName.size() > 1 && contactTwoFirstName.size() > 1) && contactOneFirstName.size() == contactTwoFirstName.size()){
+            // ToDO: Should I include code to handle this scenario?? ReturnType if Found?
+        }
+        */
+
+        if (doNamesContainMultipleParts(contactOneFirstName, contactTwoFirstName)){
+            if (isNameMatchWithHyphens(contactOneFirstName, contactTwoFirstName)) {
+                return getHyphinatedNamesMatchResults(contactOneFirstName, contactTwoFirstName); // yes or maybe
+            } else if (combineNameParts(contactOneFirstName).equals(combineNameParts(contactTwoFirstName))){
```

Comment:
Correct me if I'm wrong but this would return yes for "Mary Lou Ann" and "Maryl Ouann". That isn't what we want.

Reply:
Will add this testcase and try to think of a way to still get "D onofio" and "Donofio" to match.

Reply:
Our name part rules will make it match, won't it? Because Onofio == Onofio. If you are worried that "D'ono" would be a maybe match with "D ono" we could add a rule saying that if all the parts match exactly, the length of parts doesn't matter and will be a yes match. It would need to be an exact match though (same number of parts in both names and all parts are equal).

Reply:
As per our discussion on skype I will ensure this case will be changed to return no. However it will be possible in the future to allow the user to select it to return yes, but when doing this, I will need to add more complexity when passing through this rule. So for now this part of the code has been removed completely.